nexus/reconfigurator/planning/src/blueprint_builder/external_networking.rs

```diff
@@ -333,6 +276,139 @@ impl<'a> BuilderExternalNetworking<'a> {
     }
 }
 
+// Helper to validate that the system hasn't gone off the rails. There should
+// never be any external networking resources in the planning input (which is
+// derived from the contents of CRDB) that we don't know about from the parent
+// blueprint.
```

Comment:
I might say something explaining that this _is_ possible, but means that our planning input has been invalidated, so the caller would be expected to take another lap.

Reply:
Good call; expanded in [aca6f40](https://github.com/oxidecomputer/omicron/pull/6599/commits/aca6f40be8be6ca3c26e4ad802ef76e617446ae3)